pkg/migrate/migrate.go

```diff
@@ -41,13 +48,30 @@ var isDestScLocalVolumeProvisioner bool
 
 // Options is the set of options that should be provided to Migrate
 type Options struct {
-	SourceSCName         string
-	DestSCName           string
-	RsyncImage           string
-	Namespace            string
-	SetDefaults          bool
-	VerboseCopy          bool
-	SkipSourceValidation bool
+	SourceSCName               string
+	DestSCName                 string
+	RsyncImage                 string
+	Namespace                  string
+	SetDefaults                bool
+	VerboseCopy                bool
+	SkipSourceValidation       bool
+	SkipPVAccessModeValidation bool
+	PvcCopyTimeout             int
+	PodReadyTimeout            int
+}
+
+// PVMigrator represents a migration context for migrating data from all srcSC volumes to
+// dstSc volumes.
+type PVMigrator struct {
+	ctx             context.Context
```

Comment:
can you make ctx a function argument rather than a struct variable? it allows for more flexibility to modify the context per function call.